src/utils/sumo_utils.erl

```diff
@@ -33,7 +34,8 @@ doc_transform(Fun, Doc) ->
     FieldType = sumo_internal:field_type(Field),
     FieldName = sumo_internal:field_name(Field),
     FieldValue = sumo_internal:get_field(FieldName, Doc),
-    NewValue = Fun({FieldType, FieldName, FieldValue}),
+    FieldAttrs = sumo_internal:field_attrs(Field),
+    NewValue = Fun({FieldType, FieldName, FieldValue}, FieldAttrs),
```

Comment:
In order to give the change to handle `custom` types, remember that more info might be come in the attribute list, regarding what we discussed previously – that was the reason why we add `custom_attr()` to that list: https://github.com/inaka/sumo_db/blob/master/src/sumo.erl. And yes, we can convert it into 4-argument function.